XamlControlsGallery/ControlPagesSampleCode/SystemBackdrops/SystemBackdropsSample1.txt

```diff
@@ -0,0 +1,58 @@
+using System.Runtime.InteropServices; // For DllImport
+using WinRT; // required to support Window.As<ICompositionSupportsSystemBackdrop>()
+
+WindowsSystemDispatcherQueueHelper m_wsdqHelper; // See separate sample below for implementation
+Microsoft.UI.Composition.SystemBackdrops.MicaController m_micaController;
+Microsoft.UI.Composition.SystemBackdrops.SystemBackdropConfiguration m_configurationSource;
+
+bool TrySetMicaBackdrop()
+{
+    if (Microsoft.UI.Composition.SystemBackdrops.MicaController.IsSupported())
+    {
+        m_wsdqHelper = new WindowsSystemDispatcherQueueHelper();
+        m_wsdqHelper.EnsureWindowsSystemDispatcherQueueController();
+
+        // Hooking up the policy object
+        m_configurationSource = new Microsoft.UI.Composition.SystemBackdrops.SystemBackdropConfiguration();
+        this.Activated += Window_Activated;
+        this.Closed += Window_Closed;
+
+        // Initial configuration state.
+        m_configurationSource.IsInputActive = true;
+        switch (((FrameworkElement)this.Content).ActualTheme)
+        {
+        case ElementTheme.Dark:    m_configurationSource.Theme = Microsoft.UI.Composition.SystemBackdrops.SystemBackdropTheme.Dark; break;
+        case ElementTheme.Light:   m_configurationSource.Theme = Microsoft.UI.Composition.SystemBackdrops.SystemBackdropTheme.Light; break;
+        case ElementTheme.Default: m_configurationSource.Theme = Microsoft.UI.Composition.SystemBackdrops.SystemBackdropTheme.Default; break;
+        }
+
+        m_micaController = new Microsoft.UI.Composition.SystemBackdrops.MicaController();
+
+        // Enable the system backdrop.
+        // Note: Be sure to have "using WinRT;" to support the Window.As<...>() call.
+        m_micaController.AddSystemBackdropTarget(this.As<Microsoft.UI.Composition.ICompositionSupportsSystemBackdrop>());
+        m_micaController.SetSystemBackdropConfiguration(m_configurationSource);
+        return true; // succeeded
+    }
+
+    return false; // Mica is not supported on this system
+}
+
+private void Window_Activated(object sender, WindowActivatedEventArgs args)
+{
+    m_configurationSource.IsInputActive = args.WindowActivationState != WindowActivationState.Deactivated;
+}
+
+private void Window_Closed(object sender, WindowEventArgs args)
+{
+    // Make sure any Mica/Acrylic controller is disposed so it doesn't try to
+    // use this closed window.
+    if (m_micaController != null)
+    {
+        m_micaController.Dispose();
+        m_micaController = null;
+    }
+    this.Activated -= Window_Activated;
+    m_configurationSource = null;
+}

```

Comment:
The C++ sample closes `dispatcherQueueController` by calling `ShutdownQueueAsync` https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/system-backdrop-controller
Is this not needed for C#?

Reply:
Ideally the call would be made to ensure clean shutdown, but in practice it isn't required as long as the *Controller object is Closed/Disposed. Making the `ShutdownQueueAsync` call is non-trivial because it would require defining the COM interface in C#. The code here opts for simplicity.